functions/private/ConvertTo-Icon.ps1

```diff
@@ -2,19 +2,75 @@ function ConvertTo-Icon {
+
+        .EXAMPLE
+        try {
+            ConvertTo-Icon -bitmapPath "$env:TEMP\cttlogo.png" -iconPath "$env:TEMP\cttlogo.ico"
+        } catch [System.IO.FileNotFoundException] {
+            # Handle the thrown exception here...
+        } catch {
+            # Print out the exception to console, and see what went wrong
+            Write-Host "Could not finish 'ConvertTo-Icon' due to unhandled exception"
+            Write-Host "Error: $_"
+        }
+        This Example makes a '.ico' file at "$env:TEMP\cttlogo.ico" File Path using the bitmap file
+        found in "$env:TEMP\cttlogo.png", the function overrides the '.ico' File if it's found.
+        this function will throw a FileNotFound Exception at the event of not finding the provided bitmap File Path.
 
         .EXAMPLE
-        ConvertTo-Icon -bitmapPath "$env:TEMP\cttlogo.png" -iconPath $iconPath
+        try {
+            ConvertTo-Icon "$env:TEMP\cttlogo.png" "$env:TEMP\cttlogo.ico"
+        } catch [System.IO.FileNotFoundException] {
+            # Handle the thrown exception here...
+        } catch {
+            # Print out the exception to console, and see what went wrong
+            Write-Host "Could not finish 'ConvertTo-Icon' due to unhandled exception"
+            Write-Host "Error: $_"
+        }
+        This Example does the same as the previous one, but uses Positional Parameters instead
+
+        .EXAMPLE
+        try {
+            ConvertTo-Icon -bitmapPath "$env:TEMP\cttlogo.png" -iconPath "$env:TEMP\cttlogo.ico" -overrideIconFile $false
+        } catch [System.IO.FileNotFoundException] {
+            # Handle the thrown exception here...
+        } catch {
+            # Print out the exception to console, and see what went wrong
+            Write-Host "Could not finish 'ConvertTo-Icon' due to unhandled exception"
+            Write-Host "Error: $_"
+        }
+        This Example make use of '-overrideIconFile' Optional Parameter, the default for this paramter is $true.
+        By doing '-overrideIconFile $false', the 'ConvertTo-Icon' function will raise an exception that needs to be catched throw a 'catch' Code Block, otherwise it'll crash the running PowerShell instance/process.
+
     #>
-    param( [Parameter(Mandatory=$true)]
-        $bitmapPath,
-        $iconPath = "$env:temp\newicon.ico"
+    param(
+        [Parameter(Mandatory=$true, position=0)]
+        [string]$bitmapPath,
+        [Parameter(Mandatory=$true, position=1)]
+        [string]$iconPath,
+        [Parameter(position=2)]
+        [bool]$overrideIconFile = $true
     )
 
     Add-Type -AssemblyName System.Drawing
 
     if (Test-Path $bitmapPath) {
+        if ((Test-Path $iconPath) -AND ($overrideIconFile -eq $false)) {
+            throw "[ConvertTo-Icon] Icon File is found at '$iconPath', and the 'overrideIconFile' Parameter is set to '$overrideIconFile'. Can not continue, quitting the bitmap to icon convertion..."
```

Comment:
Shouldn't this rather be only a Warning/Info rather then a blocking errror? 
```suggestion
            Write-Host "[ConvertTo-Icon] Icon File is found at '$iconPath', and the 'overrideIconFile' Parameter is set to '$overrideIconFile'. Skipping the bitmap to icon convertion..." -ForegroundColor Yellow
            return
```

Reply:
This does make sense, as in the events of the caller of this function, if there's a `.ico`, then doing an early return is better, and somewhat safer, then doing an Exception Throw.

Thanks for the suggestion @Marterich 😄